src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final ConcurrentHashMap<EventListener, Boolean> listeners = new ConcurrentHashMap<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.put(listener, true);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners.keySet()) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown();
     }
```

Comment:
It's a good practice to also call awaitTermination after shutdown to ensure that all tasks have completed before the application exits. This is especially important if the system needs to clean up resources or save state before shutting down.